torch/distributed/pipelining/_unflatten.py

```diff
@@ -1,453 +1,8 @@
-        while node_idx < len(self.nodes):
-            node = self.nodes[node_idx]
-            assert node.op != "placeholder"
-
-            self.print()
-            self.print("STEP", node_idx, node.format_node())
-            self.print(self.module_stack)
-            if node.op == "output":
-                if len(self.module_stack) == 1:
-                    # We want the output node of the original graph to be handled
-                    # specially by the outermost stack frame (in run_outer). So
-                    # skip finalization here.
-                    return node_idx
-
-                # We've reached the end of the graph. Wrap up all the existing stack frames.
-                self.finalize_outputs()
-                return node_idx
-
-            node_module_stack = (
-                [path for path, ty in node.meta["nn_module_stack"].values()]
-                if "nn_module_stack" in node.meta
-                else self.module_stack
-            )
-            if node_module_stack[: len(self.module_stack)] != self.module_stack:
-                # This means that the current module is done executing and the
-                # current node is the beginning of a new module.
-                #
-                # In this case, we should finalize this module and return without
-                # incrementing the node counter.
-                self.finalize_outputs()
-                self.print("outlining", self.fqn)
-                self.print(self.graph)
-                return node_idx
-
-            assert node_module_stack is not None
-
-            if _is_prefix(self.module_stack, node_module_stack):
-                # This means that the current node represents the execution of a new
-                # module.
-                next_module = node_module_stack[len(self.module_stack)]
-                self.print("Creating new stack frame for", next_module)
-                # Run a nested version of module outliner from the current node
-                # counter. Once it is complete, continue from that point.
-                node_idx = _ModuleFrame(
-                    self.flat_graph,
-                    self.nodes,
-                    self.seen_nodes,
-                    self.seen_modules,
-                    self,
-                    self.module_stack + [next_module],
-                    list(node.meta["nn_module_stack"].keys())[len(self.module_stack)],
-                    self.module_call_graph,
-                ).run_from(node_idx)
-                module_idx += 1
-                continue
-
-            # The only remaining possibility is that we are in the right stack
-            # frame. Copy the node into this frame's graph and increment the node counter.
-            assert node_module_stack == self.module_stack
-            self.copy_node(node)
-            node_idx += 1
+from torch.export.unflatten import _ModuleFrame
 
 
 def _outline_submodules(orig_graph: torch.fx.Graph):
```

Comment:
Any value of this just getting moved to torch.export.unflatten as well? Or is this pretty PP specific?

Reply:
yeah, this is the one func that's special.